<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />

    <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

    <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_waypointflier_manager" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

    <!-- Namespace - all topics, services and parameters will be remapped using this namespace as a prefix (eg.: "/ros2json_bridge/odom_uav_in" to "/uav1/ros2json_bridge/odom_uav_in") -->
  <group ns="$(arg UAV_NAME)">

    <!-- Ros2jsonBridge nodelet -->
    <node pkg="nodelet" type="nodelet" name="ros2json_bridge" args="$(arg nodelet) example_ros2json_bridge/Ros2JsonBridge $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

        <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->
      <remap from="~odom_in" to="estimation_manager/odom_main" />

    </node>

  </group>

</launch>
